src/test/ripple-libpp_demo.cpp

```diff
@@ -0,0 +1,153 @@
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/st.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/basics/StringUtilities.h>
+#include <ripple/json/to_string.h>
+#include <boost/version.hpp>
+
+
+std::string serialize(ripple::STTx const& tx)
+{
+    using namespace ripple;
+
+    return strHex(tx.getSerializer().peekData());
+}
+
+std::shared_ptr<ripple::STTx const> deserialize(std::string blob)
+{
+    using namespace ripple;
+
+    auto ret{ strUnHex(blob) };
+
+    if (!ret.second || !ret.first.size())
+        Throw<std::runtime_error>("transaction not valid hex");
+
+    SerialIter sitTrans{ makeSlice(ret.first) };
+    // Can Throw
+    return std::make_shared<STTx const>(std::ref(sitTrans));
+}
+
+bool demonstrateSigning(ripple::KeyType keyType, std::string seedStr,
+    std::string expectedAccount)
+{
+    using namespace ripple;
+
+    auto const seed = parseGenericSeed(seedStr);
+    assert(seed);
+    auto const keypair = generateKeyPair(keyType, *seed);
+    auto const id = calcAccountID(keypair.first);
+    assert(toBase58(id) == expectedAccount);
+
+    STTx noopTx(ttACCOUNT_SET,
+        [&](auto& obj)
+    {
+        obj[sfAccount] = id;
+        obj[sfFee] = STAmount{ 100 };
+        obj[sfFlags] = tfFullyCanonicalSig;
+        obj[sfSequence] = 18;
+        obj[sfSigningPubKey] = keypair.first.slice();
+    });
+
+    std::cout << "\nBefore signing: \n" << noopTx << "\n"
+        << "Serialized: " << noopTx.getJson(0, true)[jss::tx] << "\n";
+
+    noopTx.sign(keypair.first, keypair.second);
+
+    auto const serialized = serialize(noopTx);
+    std::cout << "\nAfter signing: \n" << noopTx << "\n"
+        << "Serialized: " << serialized << "\n";
+
+    auto const deserialized = deserialize(serialized);
+    assert(deserialized);
+    assert(deserialized->getTransactionID() == noopTx.getTransactionID());
+    std::cout << "Deserialized: " << *deserialized << "\n";
+
+    auto const check1 = noopTx.checkSign(false);
+
+    std::cout << "Check 1: " << (check1.first ? "Good" : "Bad!") << "\n";
+    assert(check1.first);
+
+    // Use the function primitives, which are hidden by the `STTx`
+    // interface, to check the signature again and verify that
+    // `STTx` is working properly.
+    auto const& signatureSlice = noopTx[sfTxnSignature];
+    Blob const data = [&]
+    {
+        // This is a copy of the static `getSigningData` function body
+        // which is needed by `verify`.
+        Serializer s;
+        s.add32(HashPrefix::txSign);
+        noopTx.addWithoutSigningFields(s);
+        return s.getData();
+    }();
+
+    // STTx::checkSign calls `verify` indirectly via `checkSingleSign`
+    auto const check2 = verify(
+        keypair.first,
+        makeSlice(data),
+        signatureSlice,
+        true);
+
+    std::cout << "Check 2: " << (check2 ? "Good" : "Bad!") << "\n";
+
+    return check1.first && check2;
+}
+
+// Must be outside the namespace for obvious reasons
```

Comment:
Not sure this comment makes sense since we never opened a namespace in this file?